daemon/start.go

```diff
@@ -265,3 +275,10 @@ func (daemon *Daemon) Cleanup(container *container.Container) {
 		logrus.Errorf("%s cleanup: failed to delete container from containerd: %v", container.ID, err)
 	}
 }
+
+func withImageName(imageName string) containerd.NewContainerOpts {
```

Comment:
Do you think it would make sense to upstream this? https://github.com/containerd/containerd/blob/2f60e389a03740339c9c2762004fdcb5de489b09/container_opts.go#L67